ReactiveCocoa/NSObject+KeyValueObserving.swift

```diff
@@ -106,6 +106,22 @@ extension Reactive where Base: NSObject {
 		return producer(for: keyPath) { $0 as! U? }
 	}
 
+	/// Create a producer which sends the current value and all the subsequent
+	/// changes of the property specified by the key path.
+	///
+	/// The producer completes when the object deinitializes.
+	///
+	/// - parameters:
+	///   - keyPath: The key path of the property to be observed.
+	///
+	/// - returns: A producer emitting values of the property specified by the
+	///            key path.
+	public func producer<U>(for keyPath: KeyPath<Base, U?>) -> SignalProducer<U?, Never> where U: RawRepresentable {
+		return producer(for: keyPath) {
+			$0.flatMap { value in U(rawValue: value as! U.RawValue) }
```

Comment:
I'm not sure if this should force-unwrap the enum value or not. Strictly speaking, if we have a value, we should be able to create a valid enum instance from it. But there's also no "harm" in returning the optional here since we have to return an optional anyway.

Reply:
This should trap in my opinion. If a type is imported as a Swift enum equivalent, it should be exactly convertible in both directions. Most importantly, given that the property is optional, `nil` has its own contextual semantic, and we should not silently corrupt that semantic with the scenario of "failed conversion that is supposed to succeed".